contributors/design-proposals/kubelet-authorizer.md

```diff
@@ -0,0 +1,119 @@
+  // and may be empty if a specific node cannot be determined.
+  // isNode is true if the user.Info represents an identity issued to a node.
+  IdentifyNode(user.Info) (nodeName string, isNode bool)
+}
+```
+
+The default `NodeIdentifier` implementation:
+* `isNode` - true if the user groups contain the `system:nodes` group
+* `nodeName` - populated if `isNode` is true, and the user name is in the format `system:node:<nodeName>`
+
+## Node authorizer
+
+A new node authorizer will be inserted into the authorization chain:
+* API server authorizer (authorizes "loopback" API clients used by components within the API server)
+* Node authorizer (new)
+* User-configured authorizers... (e.g. ABAC, RBAC, Webhook)
+
+The node authorizer is responsible for ensuring:
+* The request is made by a node (`IdentifyNode()` returns isNode=true)
+* The request is allowed by the rules contained in the default `system:node` cluster role
+
+Requests that do not meet those conditions are forbidden by this authorizer.
+Subsequent authorizers in the chain can run and choose to allow the request.
+
+Requests from identifiable nodes (`IdentifyNode()` returns nodeName != "") for secrets and configmaps are further restricted:
+* Requests for secrets are limited to `get`, and the requested secret must be related to the requesting node by one of the following relationships:
+  * node -> pod -> secret
+  * node -> pod -> pvc -> pv -> secret
+  * node -> pod -> pvc -> storageclass -> secret
+* Requests for configmaps are limited to `get`, and the requested configmap must be related to the requesting node by one of the following relationships:
+  * node -> pod -> configmap
+
+## Node admission
+
+A new node admission plugin is made available that enforces the following:
+
+Limits `create` of node resources by identifiable nodes:
+* only allow the node object corresponding to the node making the API request
+
+Limits `create` of pod resources by identifiable nodes:
+* only allow pods with mirror pod annotations
+* only allow pods with nodeName set to the node making the API request
+* only allow pods with no secret references
+
+Limits `update`,`patch`,`delete` of node and nodes/status resources by identifiable nodes:
+* only allow modifying the node object corresponding to the node making the API request
+
+Limits `update`,`patch`,`delete` of pod and pod/status resources by identifiable nodes:
+* only allow modifying pods with nodeName set to the node making the API request (requires fetching the pod on delete)
+* do not allow removing a mirror pod annotation
+
+## API Changes
+
+None
+
+## RBAC Changes
+
+Currently, the `system:node` cluster role is automatically bound to the `system:nodes` group.
+
+Because the node authorizer accomplishes the same purpose, with the benefit of additional restrictions
+on secret and configmap access, this binding is no longer needed, and will no longer be set up automatically.
```

Comment:
It would also need to do this for _some_ APIs for all users in the `system:nodes` group, not just those without a node-specific identity, right? I'm talking about things like "list services" and "get endpoints".